consoleme/lib/oidc.py

```diff
@@ -82,6 +83,20 @@ async def populate_oidc_config():
     return oidc_config
 
 
+async def validate_redirect_uri(request, redirect_uri: str):
+    """Ensure a redirect URI begins with the ConsoleMe server URL"""
+    protocol = request.request.protocol
+    if "https://" in config.get("url"):
+        # If we're behind a load balancer that terminates tls for us, request.request.protocol will be "http://" and our
+        # oidc redirect will be invalid
+        protocol = "https"
+    base_url = f"{protocol}://{request.request.host}"
```

Comment:
I don't think we should be building the validation off the host field, this assumes we receive a known good host header. I think this should be configuration driven instead.